sui/p2p_transfer/index.js

```diff
@@ -19,48 +20,47 @@ const main = async () => {
   const SENDER_PRIVATE_KEY = process.env.ACC1_PRIVATE_KEY;
   const sender_keypair = getKeyPairFromExportedPrivateKey(SENDER_PRIVATE_KEY);
 
-  const RECIEVER_PRIVATE_KEY = process.env.ACC2_PRIVATE_KEY;
-  const receiver_keypair = getKeyPairFromExportedPrivateKey(RECIEVER_PRIVATE_KEY);
-  const receiver_address = receiver_keypair.getPublicKey().toSuiAddress();
-
 
   // create a new SuiClient object pointing to the network you want to use
   let url = getFullnodeUrl('mainnet');
   if (URL_OVERRIDE) {
       url = URL_OVERRIDE;
   }
   const suiClient = new SuiClient({ url: url });
+  let gasCoin = null
 
   while (true) {
     try {
       const txb = new TransactionBlock();
       const [coin] = txb.splitCoins(txb.gas, [txb.pure(1)]);
-      txb.transferObjects([coin], receiver_address);
+      txb.transferObjects([coin], sender_keypair.toSuiAddress());
+      txb.setSender(sender_keypair.toSuiAddress());
+      txb.setGasBudget(5_000_000)
+
+      // This doesn't change e2e latency, but is recommended for saving an extra rpc call during the build phase
+      if (gasCoin) {
+        txb.setGasPayment([gasCoin]);
```

Comment:
If we want to avoid these kinds of optimizations, I'm happy to remove this.  This is how a lot of apps currently are structured, but doing this (currently) requires this extra step.  It only affects the build metric, and not the e2e latency numbers